hysds_commons/job_utils.py

```diff
@@ -390,7 +414,11 @@ def resolve_hysds_job(job_type=None, queue=None, priority=None, tags=None, param
         # match_inputs(param,context)
         logger.info("param: {}".format(param))
         if param["name"] not in params:
-            raise RuntimeError("'params' must specify '{0}' parameter".format(param["name"]))
+            if param["name"] in optional_params:
+                logger.warning("{0} is not given but optional, skipping...".format(param["name"]))
```

Comment:
Per the Jira ticket, if an optional parameter was not specified in the `params`, should we additionally check if there is a default value specified in the job spec and add it to the job payload?

Reply:
good catch ill change it to use the default value is given and skip if not given

Reply:
Sounds good. Note the different default types that can be possible (i.e. string, boolean, int, etc): https://hysds-core.atlassian.net/wiki/spaces/HYS/pages/199787836/Job+and+HySDS+IO+Specifications#Valid-Param-Types

Reply:
through the Rest API do we pass them in as their proper types (ex. `number`, `text`, `enum`) or are they passed in as strings? just want to make sure

Reply:
that's a good question. I actually don't know.

Reply:
@DustinKLo: how are they being passed through right now? I think they're all strings (correct me if I'm wrong).

Reply:
@pymonger according to the `hysds-io` schema: https://github.com/hysds/hysds_commons/blob/develop/schemas/hysds-io-schema.json#L115-L134
the types can only be `string` unless specified as `objects`

```
Validating hysds-io for test_params
[ERROR] name: test_params, location: deque(['params', 2, 'default']), message: 1337 is not of type 'string'
```

i think that's a little outdated, maybe we should update the schema to accept number values so we don't have to coerce it to a `float` type 
https://github.com/hysds/hysds_commons/blob/develop/hysds_commons/job_utils.py#L91-L92

Reply:
`resolve_hysds_job` will now handle the `hysdsios` inputs and `default` types
its a little redundant b/c `resolve_mozart_job` also handles but i wanted to avoid refactoring a bunch of code